File: js/examples/basic-rest.ts
import { MetaScalpClient } from '../src';

async function main() {
  // Discover running MetaScalp instance
  const client = await MetaScalpClient.discover();
  console.log(`Connected to MetaScalp on port ${client.port}`);

  // List connections
  const { connections } = await client.getConnections();
  console.log(`Found ${connections.length} connection(s)`);

  for (const conn of connections) {
    console.log(`  [${conn.id}] ${conn.name} — ${conn.exchange} ${conn.market} (state: ${conn.state})`);

    // Get tickers
    const { tickers } = await client.getTickers(conn.id);
    console.log(`  ${tickers.length} tickers available`);

    // Get balance
    const { balances } = await client.getBalance(conn.id);
    for (const b of balances) {
      if (b.total > 0) {
        console.log(`  Balance: ${b.coin} — total: ${b.total}, free: ${b.free}, locked: ${b.locked}`);
      }
    }
  }
}

main().catch(console.error);

File: python/examples/basic_rest.py
"""Basic REST API usage example."""

import asyncio
from metascalp import MetaScalpClient


async def main():
    # Discover running MetaScalp instance
    client = await MetaScalpClient.discover()
    print(f"Connected to MetaScalp on port {client.port}")

    # List connections
    data = await client.get_connections()
    connections = data["connections"]
    print(f"Found {len(connections)} connection(s)")

    for conn in connections:
        print(f"  [{conn['id']}] {conn['name']} — {conn['exchange']} {conn['market']}")

        # Get balance
        balance_data = await client.get_balance(conn["id"])
        for b in balance_data["balances"]:
            if b["total"] > 0:
                print(f"    {b['coin']}: total={b['total']}, free={b['free']}, locked={b['locked']}")

    await client.close()


asyncio.run(main())
